planner/core/rule_partition_processor.go

```diff
@@ -94,36 +95,20 @@ func (s *partitionProcessor) rewriteDataSource(lp LogicalPlan) (LogicalPlan, err
 // partitionTable is for those tables which implement partition.
 type partitionTable interface {
 	PartitionExpr(ctx sessionctx.Context, columns []*expression.Column, names types.NameSlice) (*tables.PartitionExpr, error)
-}
-
-func generateHashPartitionExpr(t table.Table, ctx sessionctx.Context, columns []*expression.Column, names types.NameSlice) (*tables.PartitionExpr, error) {
-	tblInfo := t.Meta()
-	pi := tblInfo.Partition
-	var column *expression.Column
-	schema := expression.NewSchema(columns...)
-	exprs, err := expression.ParseSimpleExprsWithNames(ctx, pi.Expr, schema, names)
-	if err != nil {
-		return nil, err
-	}
-	exprs[0].HashCode(ctx.GetSessionVars().StmtCtx)
-	if col, ok := exprs[0].(*expression.Column); ok {
-		column = col
-	}
-	return &tables.PartitionExpr{
-		Column: column,
-		Expr:   exprs[0],
-		Ranges: nil,
-	}, nil
+	GetOriginPartitionExpr() ast.ExprNode
```

Comment:
We don't need to add new methods to the interface.
`PartitionExpr().OrigExpr` works the same.
If you'd like to, maybe wrap a local function is better.